api/src/Security/Authenticator/HttpBearerAuthenticator.php

```diff
@@ -0,0 +1,159 @@
+
+/**
+ * @author Vincent Chalamon <vincentchalamon@gmail.com>
+ *
+ * @final
+ */
+class HttpBearerAuthenticator implements AuthenticatorInterface, AuthenticationEntryPointInterface
+{
+    /**
+     * @var array<string, class-string<Signer>>
+     */
+    public const SIGN_ALGORITHMS = [
+        'HS256' => Signer\Hmac\Sha256::class,
+        'HS384' => Signer\Hmac\Sha384::class,
+        'HS512' => Signer\Hmac\Sha512::class,
+        'ES256' => Signer\Ecdsa\Sha256::class,
+        'ES384' => Signer\Ecdsa\Sha384::class,
+        'ES512' => Signer\Ecdsa\Sha512::class,
+        'RS256' => Signer\Rsa\Sha256::class,
+        'RS384' => Signer\Rsa\Sha384::class,
+        'RS512' => Signer\Rsa\Sha512::class,
+    ];
+
+    private UserProviderInterface $userProvider;
+    private string $realmName;
+    private string $payloadKey;
+    private ?string $key;
+    private Signer $signer;
+    private HttpClientInterface $client;
+    private LoggerInterface $logger;
+
+    public function __construct(UserProviderInterface $userProvider, string $realmName, string $payloadKey, string $algorithm, string $publicKey, HttpClientInterface $client = null, LoggerInterface $logger = null)
+    {
+        $this->userProvider = $userProvider;
+        $this->realmName = $realmName;
+        $this->payloadKey = $payloadKey;
+        $signerClass = self::SIGN_ALGORITHMS[$algorithm];
+        $this->signer = new $signerClass();
+        $this->key = $publicKey;
+        $this->client = $client ?: HttpClient::create();
+        $this->logger = $logger;
+    }
+
+    public function start(Request $request, AuthenticationException $authException = null): Response
+    {
+        $response = new Response();
+        $response->headers->set('WWW-Authenticate', sprintf('Bearer realm="%s"', $this->realmName));
+        $response->setStatusCode(401);
+
+        return $response;
+    }
+
+    public function supports(Request $request): ?bool
+    {
+        return $request->headers->has('AUTHORIZATION');
+    }
+
+    public function authenticate(Request $request): Passport
+    {
+        if (!class_exists(Configuration::class)) {
+            throw new RuntimeException('HttpBearerAuthenticator requires lcobucci/jwt, please run "composer require lcobucci/jwt" to install it.');
+        }
+
+        try {
+            $configuration = Configuration::forSymmetricSigner($this->signer, $this->getKey($this->key));
+
+            // Decode token
+            $token = $configuration->parser()->parse(substr($request->headers->get('AUTHORIZATION'), 7));
+
+            // Validate token
+            $configuration->setValidationConstraints(new SignedWith($configuration->signer(), $configuration->signingKey()));
+            $configuration->validator()->assert($token, ...$configuration->validationConstraints());
+        } catch (Exception $exception) {
+            throw new AuthenticationException($exception->getMessage(), $exception->getCode(), $exception);
+        }
+
```

Comment:
This could be nice to provide an extension point to do some extra check (such as calling the introspection API of an OIDC server).

Reply:
Like dispatching an event? Or just allowing the developer to override the method in its own authenticator and do whatever he/she wants?

Reply:
Extending the authenticator is not trivial as one has to inject the right user provider etc.. better use events for this. 
Maybe just tell people to rely on the existing [CheckPassportEvent](https://github.com/symfony/symfony/blob/6.1/src/Symfony/Component/Security/Http/Event/CheckPassportEvent.php) whose purpose is precisely to perform extra checks on the credentials/user (see e.g. https://github.com/symfony/symfony/blob/6.1/src/Symfony/Component/Security/Http/EventListener/CheckCredentialsListener.php)